macros/sfdc_formula_refactor.sql

```diff
@@ -0,0 +1,38 @@
+{%- macro sfdc_formula_refactor(join_to_table) -%}
+
+    --Generate the key value pair from the formula field table with the below macro.
+    {%- set key_val = salesforce_formula_utils.sfdc_get_formula_column_values(source('salesforce', 'fivetran_formula'), 'field', 'sql', join_to_table) -%}
+
+    --Creating a permanent array to be carried over after this macro.
+    {%- set adjusted_key_val = [] -%}
+
+    {%- for key, value in key_val %}
+        
+        {%- set temp_array = [] -%} --Creating a temporary array for the key (field name) and value (sql) pairs to be created within.
+        {{ temp_array.append(key)|default("", True)  }} --The key will always be appended without additional adjustments.
+
+        --If the field name is found within the sql of another field. Then an adjustment to the sql will take place
+        --The adjustment will replace the field name with the true sql of the other formual field.
+        {%- for field, sql in key_val -%}
+            {%- if field in value -%}
+                {{ temp_array.append( value | replace(field,sql) )| default("", True) }}
```

Comment:
we should probably encase the replaced sql with parentheses right? #pemdas lol

Reply:
Great catch! I can't believe I forgot to please excuse my dear aunt sally. Just pushed a commit with this pemdas taken into consideration by adding quotes.